src/main/webapp/app/entities/iris/iris-message.model.ts

```diff
@@ -0,0 +1,34 @@
+import { BaseEntity } from 'app/shared/model/base-entity';
+import { IrisMessageContent } from 'app/entities/iris/iris-content-type.model';
+import dayjs from 'dayjs';
+
+export enum IrisSender {
+    LLM = 'LLM',
+    USER = 'USER',
+    ARTEMIS = 'ARTEMIS',
+}
+
+export class IrisServerMessage implements BaseEntity {
+    id: number;
+    content: IrisMessageContent[];
+    sentAt: dayjs.Dayjs;
+    sender: IrisSender.LLM | IrisSender.ARTEMIS;
+    helpful?: boolean;
+}
+
+export class IrisClientMessage implements BaseEntity {
+    id?: number;
+    content: IrisMessageContent[];
+    sentAt?: dayjs.Dayjs;
+    sender: IrisSender.USER;
+}
+
+export type IrisMessage = IrisServerMessage | IrisClientMessage;
```

Comment:
Since I did not get an answer on your closed PR:
Any reason why you don't just use inheritance here?

Reply:
The use of inheritance is beneficial when you have a hierarchical relationship between classes, and child classes share common behavior and structure with the parent class but also have their own specialized behavior. In this case, we have a similar structure but do not have any encapsulated behavior. To implement inheritance here, one should introduce a superclass with all common fields, while subclasses have to redefine nullable fields from the superclass that are not null, which would increase the size of the codebase without bringing any real benefits.